### PR TITLE
unify http plaintext tls tests

### PR DESCRIPTION
**Pull request by guyarb**
### What does this PR do?

Runs most of the HTTP tests for plaintext and TLS

### Motivation

Expand the coverage of tests for go-tls

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by DanielLavie — CHANGES_REQUESTED** [review]

File: pkg/network/usm/monitor_test.go [hunk 1]
```diff
@@ -411,11 +411,15 @@ func (s *httpTestSuite) TestRSTPacketRegression() {
 	c.Close()
 	time.Sleep(100 * time.Millisecond)
```
[the remark this answers is not in the document]
Comment (guyarb) [reply pkg/network/usm/monitor_test.go:488]: yes, the sleep is meant to trigger the RST on the connection and let USM process it

File: pkg/network/usm/monitor_tls_test.go [hunk 2]
```diff
@@ -776,7 +777,10 @@ func setupUSMTLSMonitor(t *testing.T, cfg *config.Config) *Monitor {
 	usmMonitor, err := NewMonitor(cfg, nil, nil, nil)
 	require.NoError(t, err)
 	require.NoError(t, usmMonitor.Start())
-	t.Cleanup(usmMonitor.Stop)
+	t.Cleanup(func() {
+		usmMonitor.Stop()
+		libtelemetry.Clear()
+	})
 	t.Cleanup(utils.ResetDebugger)
```
[the remark this answers is not in the document]
Comment (guyarb) [reply pkg/network/usm/monitor_tls_test.go:784]: you're right

**Review by DanielLavie — APPROVED** [review]

**Pull request closed without merging**